Codes/Section_05_VEC_Labeling_for_Tables/tika_text_extractor.py

```diff
@@ -1,43 +1,103 @@
-    """This function takes the extracted text and stores it in our MSSQL database."""
+def split_sents_with_bullet_points(sents_list):
+    """This function splits a list of sentences into a list of lists of sentences, where each list contains sentences
+    that are separated by a bullet point."""
+    new_sents_list = []
+    d = '•'
+    for sent in sents_list:
+        if (d in sent) == True:
+            new_sents = [d+e for e in sent.split(d) if e]
+            for sent in new_sents:
+                if len(sent) > 6:
+                    new_sents_list.append(sent)
+        else:
+            new_sents_list.append(sent)
+    return new_sents_list
 
-# run script in root directory
-pdfs_path = str(Path('.') / "Data_Files/esa-project-pdfs-subsample")
-output_path = pdfs_path / "extracted_texts"
+def regex_remove_items_with_page_number(sents_list):
+    """This function removes all sentences that contain a page number."""
+    new_sents_list = []
+    for sent in sents_list:
+        if re.search(r'\d+\s*[pP]\s*\d+', sent) == None:
+            new_sents_list.append(sent)
+    return new_sents_list
+
+## TODO~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+def regex_remove_spaced_words(sents_list):
+    """This function removes all sentences where there is a space between every single character.
+    It checks if there are more then 4 spaces in the first 10 characters."""
+    new_sents_list = []
+    for sent in sents_list:
+        if re.search(r'\s{4,}', sent[:10]) == None:
+            new_sents_list.append(sent)
+    return new_sents_list
 
-if not output_path.exists():
-    output_path.mkdir()
+
+if not path.exists(output_path):
+    makedirs(output_path)
 
 def process_file(filename):
-    in_filename = str(Path('.') / filename)
-    out_filename = str(Path('.') / 'extracted_texts' / filename) + '.txt'
+    """This function takes a single PDF file and extracts the text from it."""
+    in_filename = pdfs_path + '/' + filename
+    out_filename = output_path + '/' + filename[:-4] + '.txt'
+    pickle_filename = pickled_sentences_path + '/' + filename[:-4] + '.pkl'
 
     text = parser.from_file(in_filename)
     with open(out_filename, 'w+') as outfile:
-        # outfile.write(text["content"])
-        print(text['content'])
+        outfile.write(text["content"])
+        # print(text['content'])
+        doc = nlp(text['content'])
+        sents_list = []
+        for sent in doc.sents:
+            sents_list.append(sent.text)
+        sents_list = remove_false_sentences(sents_list)
+        sents_list = split_sents_with_bullet_points(sents_list)
+        print(sents_list)
+
+        # pickle.dump(sents_list, open(pickle_filename, 'wb'))
 
 if __name__ == '__main__':
     time_start = time.time()
     pool = mp.Pool()
-    pool.map(process_file, list_files(pdfs_path))
+    pool.map(process_file, list_files(pdfs_path)[0:4])
```

Comment:
Magic number: why 4?

Reply:
This was just for checking if the extraction works. So, I had the code only run through the first few examples.

Note: I am no longer using this code and will be removing it from the repository once I get to Section 05. However, I've updated text extraction code for another project and may add it somewhere else in berdi, but not sure.